bus-mapping/src/evm/opcodes/returndatacopy.rs

```diff
@@ -41,10 +74,115 @@ impl Opcode for Returndatacopy {
                 // there is no more steps.
             }
         }
+
+        let copy_event = gen_copy_event(state, geth_step)?;
+        state.push_copy(copy_event);
         Ok(vec![exec_step])
     }
 }
 
+fn gen_copy_steps(
+    state: &mut CircuitInputStateRef,
+    exec_step: &mut ExecStep,
+    src_addr: u64,
+    dst_addr: u64,
+    src_addr_end: u64,
+    bytes_left: u64,
+    _is_root: bool,
+) -> Result<Vec<CopyStep>, Error> {
+    let mut copy_steps = Vec::with_capacity(2 * bytes_left as usize);
+    for idx in 0..bytes_left {
+        let addr = src_addr + idx;
+        let rwc = state.block_ctx.rwc;
+        let (value, is_pad) = if addr < src_addr_end {
+            let byte = state.call_ctx()?.return_data
+                [(addr - state.call()?.last_callee_return_data_offset) as usize];
+            state.push_op(
+                exec_step,
+                RW::READ,
+                MemoryOp::new(state.call()?.last_callee_id, addr.into(), byte),
+            );
+            (byte, false)
+        } else {
+            //TODO: return out of bound
+            assert!(addr < src_addr_end, "return data copy out of bound");
+            (1, false)
```

Comment:
Will this ever be reached? Otherwise `unreachable!` or something may make more sense.

Reply:
fixed